### DISCUSSION TITLE
Advice for performance space to play pre-recorded video to a schedule

### DISCUSSION THREAD

**Discussion by davegoopot** [mozilla/hubs#2829] [General Questions & Discussion] [answered]
Hi, 

Very new here so apologies if this is covered elsewhere. I'm working on setting up a Hub to work as a stage for some local musicians to "perform" their work over a weekend. The content will be video all recorded in advance. I'm thinking the look will be very similar to the "Hubs Commons" room here https://hubs.mozilla.com/scenes/T5QUL3L/hubs-commons . I have two questions, below. I'd also be grateful of any general advice and tips.

1. I'd like to add a video player object into the room and be able to swap in and play links to different videos in the course of the weekend. What's the best way to do that? Do I put the video in the Spoke scene in advance, or do I need to make it on the fly in the Hub?

2. YouTube would be the preferred platform for hosting the video, but I've seen a post that this might be blocked. Is that a real problem? Any suggestions on alternatives? (See https://github.com/mozilla/hubs/discussions/2681 )

**Comment by brianpeiris** [comment]
Hi @davegoopot, unfortunately there isn't an easy way to switch videos out in the middle of an event. We have a feature planned that should help with this, you can follow along on that here: https://github.com/mozilla/hubs/issues/1324. In the meantime, other options might be to live stream a feed from a source that you control, for example from [OBS](https://obsproject.com/) so that you can switch the feed at the source, instead of in Hubs. You could stream from OBS to Twitch or YouTube, though you'll probably have better luck with Twitch, due to YouTube's restrictions. 

The particular videos you mentioned in #2681 were probably failing due to the limits that YouTube imposes on the number of videos we can stream from them during a certain time period, so unfortunately YouTube will sometimes be inconsistent in Hubs.

Lastly, you could also consider streaming via a paid streaming service. In general Hubs is able to play any video stream that is publicly available on the Web. So if you are able to setup your own stream on a server you own or a service you pay for, it is more likely to be reliable.

**Reply by davegoopot** [reply]
I've not tried this, but if I have two Spoke scenes with different videos, but otherwise identical, can I swap the Hub scene and the video change?

**Reply by tedtheminer** [reply]
@davegoopot I've tried that and if you change it in Spoke and publish it again it doesn't change it for others unless they refresh the page

**Comment by ocularvr** [comment, marked as answer]
Quick solution: 

1. Go to (for example) https://tinycc.com
2. Make short url for your youtube video, like http://6m5d.2.vu/1
3. In Spoke, make the video source the short link
4. >>different videos in the course of the weekend.<< Keep the same url, just change the short link in tinycc. This way, your videos change without you having to do anything in Spoke. 

Benefits - no need to change in spoke, no need to republish, saves a lot of time. Keep your hubs url the same and only modify the url in tinycc. Much easier for me. 

Longer term solution

Make some javascript that can do this and we can set the dates/times/videos in advance, then we could upload it to our hubs server
